contracts/BatchExchange.sol

```diff
@@ -572,7 +577,19 @@ contract BatchExchange is EpochTokenLocker {
       * @param executedAmount proportion of order's requested sellAmount that was filled.
       */
     function revertRemainingOrder(address owner, uint16 orderId, uint128 executedAmount) private {
-        orders[owner][orderId].usedAmount = orders[owner][orderId].usedAmount.sub(executedAmount).toUint128();
+        if (isOrderTrackingItsUsedAmount(orders[owner][orderId])) {
+            orders[owner][orderId].usedAmount = orders[owner][orderId].usedAmount.sub(executedAmount).toUint128();
+        }
+    }
+
+    /** @dev Checks whether an order is intended to track its usedAmount
+      * @param order order under inspection
+      * @return true if the given order does track its usedAmount
+      */
+    function isOrderTrackingItsUsedAmount(Order memory order) private pure returns (bool) {
+        return
+            order.priceNumerator < PRICE_NUMERATOR_OR_DENOMINATOR_FOR_ORDERS_NOT_TRACKING_USED_AMOUNTS &&
+            order.priceDenominator < PRICE_NUMERATOR_OR_DENOMINATOR_FOR_ORDERS_NOT_TRACKING_USED_AMOUNTS;
```

Comment:
Is there a reason you prefer `<` here instead of `!=`? I feel like `!=` would more clearly indicate that its a "magic number".